packages/rolldown/build.ts

```diff
@@ -103,30 +107,62 @@ function withShared(
       // some dependencies, e.g. zod, cannot be inlined because their types
       // are used in public APIs
       ...Object.keys(pkgJson.dependencies ?? {}),
-      bindingFileWasi,
-      bindingFileWasiBrowser,
     ],
     define: {
       'import.meta.browserBuild': String(isBrowserBuild),
     },
     ...options,
     plugins: [
+      isBrowserPkg && resolveWasiBinding(isBrowserBuild),
       isBrowserBuild && removeBuiltModules(),
       options.plugins,
     ],
   };
 }
+function resolveWasiBinding(isBrowserBuild?: boolean): Plugin {
+  return {
+    name: 'resolve-wasi-binding',
+    resolveId: {
+      filter: { id: /\bbinding\b/ },
+      async handler(id, importer, options) {
```

Comment:
Could you add some comments for this plugin? About what it does and the contexts.

Reply:
Done.